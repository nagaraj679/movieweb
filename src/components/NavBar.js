import React from 'react';


//I use a class component because I need to add some logic inside (burgerToggle) 
class NavBar extends React.Component {
  constructor(props) {
    super();
    this.burgerToggle = this.burgerToggle.bind(this);
  }
  
  burgerToggle() {
    const links = document.querySelector('.narrow-links');
    links.style.display = links.style.display == 'block' ? 'none' : 'block';
  }

  render() {
    return (
      <nav>
        <div className="nav-wide">
          <div className="wide">
            <a href="#">Cinema</a>
            <a href="#">Movies</a>
            <a href="#">News</a>
          </div>
        </div>
        <div className="nav-narrow"> 
          <i className="fa fa-bars fa-2x" onClick={this.burgerToggle}></i>
          <div className="narrow-links">
            <a href="#" onClick={this.burgerToggle}>Cinema</a>
            <a href="#" onClick={this.burgerToggle}>Movies</a>
            <a href="#" onClick={this.burgerToggle}>News</a>
          </div>
        </div>
      </nav>
    )
  }
}

export default NavBar;


