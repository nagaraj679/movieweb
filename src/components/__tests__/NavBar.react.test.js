import React from 'react';
import { shallow, mount, render } from 'enzyme';
import NavBar from '../NavBar';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

describe('NavBar', () => {

    it('should be defined', () => {
      expect(NavBar).toBeDefined();
    });

    it('should render correctly', () => {
      const tree = shallow(
        <NavBar/>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should contain 6 link', () => {
      const tree = shallow(
        <NavBar/>
      );
      expect(tree.find('a')).toHaveLength(6);
    })

});