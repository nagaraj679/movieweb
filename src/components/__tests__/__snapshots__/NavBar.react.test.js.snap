// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="nav-wide"
        >
          <div
            className="wide"
          >
            <a
              href="#"
            >
              Cinema
            </a>
            <a
              href="#"
            >
              Movies
            </a>
            <a
              href="#"
            >
              News
            </a>
          </div>
        </div>,
        <div
          className="nav-narrow"
        >
          <i
            className="fa fa-bars fa-2x"
            onClick={[Function]}
          />
          <div
            className="narrow-links"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              Cinema
            </a>
            <a
              href="#"
              onClick={[Function]}
            >
              Movies
            </a>
            <a
              href="#"
              onClick={[Function]}
            >
              News
            </a>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="wide"
          >
            <a
              href="#"
            >
              Cinema
            </a>
            <a
              href="#"
            >
              Movies
            </a>
            <a
              href="#"
            >
              News
            </a>
          </div>,
          "className": "nav-wide",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href="#"
              >
                Cinema
              </a>,
              <a
                href="#"
              >
                Movies
              </a>,
              <a
                href="#"
              >
                News
              </a>,
            ],
            "className": "wide",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cinema",
                "href": "#",
              },
              "ref": null,
              "rendered": "Cinema",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Movies",
                "href": "#",
              },
              "ref": null,
              "rendered": "Movies",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "News",
                "href": "#",
              },
              "ref": null,
              "rendered": "News",
              "type": "a",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fa fa-bars fa-2x"
              onClick={[Function]}
            />,
            <div
              className="narrow-links"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                Cinema
              </a>
              <a
                href="#"
                onClick={[Function]}
              >
                Movies
              </a>
              <a
                href="#"
                onClick={[Function]}
              >
                News
              </a>
            </div>,
          ],
          "className": "nav-narrow",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-bars fa-2x",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  href="#"
                  onClick={[Function]}
                >
                  Cinema
                </a>,
                <a
                  href="#"
                  onClick={[Function]}
                >
                  Movies
                </a>,
                <a
                  href="#"
                  onClick={[Function]}
                >
                  News
                </a>,
              ],
              "className": "narrow-links",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cinema",
                  "href": "#",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Cinema",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Movies",
                  "href": "#",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Movies",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "News",
                  "href": "#",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "News",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="nav-wide"
          >
            <div
              className="wide"
            >
              <a
                href="#"
              >
                Cinema
              </a>
              <a
                href="#"
              >
                Movies
              </a>
              <a
                href="#"
              >
                News
              </a>
            </div>
          </div>,
          <div
            className="nav-narrow"
          >
            <i
              className="fa fa-bars fa-2x"
              onClick={[Function]}
            />
            <div
              className="narrow-links"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                Cinema
              </a>
              <a
                href="#"
                onClick={[Function]}
              >
                Movies
              </a>
              <a
                href="#"
                onClick={[Function]}
              >
                News
              </a>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="wide"
            >
              <a
                href="#"
              >
                Cinema
              </a>
              <a
                href="#"
              >
                Movies
              </a>
              <a
                href="#"
              >
                News
              </a>
            </div>,
            "className": "nav-wide",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  href="#"
                >
                  Cinema
                </a>,
                <a
                  href="#"
                >
                  Movies
                </a>,
                <a
                  href="#"
                >
                  News
                </a>,
              ],
              "className": "wide",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cinema",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Cinema",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Movies",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Movies",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "News",
                  "href": "#",
                },
                "ref": null,
                "rendered": "News",
                "type": "a",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fa fa-bars fa-2x"
                onClick={[Function]}
              />,
              <div
                className="narrow-links"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  Cinema
                </a>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  Movies
                </a>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  News
                </a>
              </div>,
            ],
            "className": "nav-narrow",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-bars fa-2x",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    Cinema
                  </a>,
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    Movies
                  </a>,
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    News
                  </a>,
                ],
                "className": "narrow-links",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Cinema",
                    "href": "#",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Cinema",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Movies",
                    "href": "#",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Movies",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "News",
                    "href": "#",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "News",
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
